### PR TITLE
Auto register ide-helper hook using Package Discovery

### PR DESCRIPTION
**Pull request by daniel-de-wit**
Would you consider automatically registering the model hook using Package Discovery?

### CODE REVIEW

**Comment by staudenmeir** [comment]
Hi @daniel-de-wit,
Thanks, I'll take a look.

**Comment by daniel-de-wit** [comment]
@staudenmeir I've updated the PR to make the service provider deferred, just like the one from IDEHelper. This ensures the order of package discovery doesn't effect the registration of the model hook. Now it only registers after the ModelsCommand is requested from the service container, which removed the need for a couple of environment guards.

I would like to hear your take on it

File: src/Providers/EloquentHasManyDeepServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    public function register(): void
+    {
+        $this->registerConfig();
+        $this->registerIDEHelperHook();
+    }
+
+    public function provides(): array
+    {
+        return [
+            ModelsCommand::class,
```
Comment (staudenmeir) [inline src/Providers/EloquentHasManyDeepServiceProvider.php:27]: Why is it necessary to add this command here?

**Comment by staudenmeir** [comment]
Thanks, excellent PR! I've released a new version.

**Comment by staudenmeir** [comment]
One last question: You kept and even improved the README section about manually enabling the model hook.

Do you still see a use case for that?

**Comment by daniel-de-wit** [comment]
I added that for people that have package discovery disabled. I never encountered a project that disabled the entire package discovery though.

**Comment by staudenmeir** [comment]
Ok, thanks.

**Pull request merged**